.github/actions/env/action.yml

```diff
@@ -0,0 +1,48 @@
+name: Environment setup
+
+on:
+  workflow_dispatch:
+    inputs:
+      python-version:
+        description: "The python version desired."
+        required: true
+        default: '3.7'
+        type: string
+
+      pytorch-version:
+        description: "The pytorch version desired."
+        required: true
+        default: '1.10.2'
+        type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup environment (conda)
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        auto-update-conda: true
+        python-version: ${{ inputs.python-version }}
+
+    - if: ${{ contains(fromJson('["lts", "nightly"]'), inputs.pytorch-version ) }}
+      name: Install dependencies
+      shell: bash -l {0}
+      run: |
+        conda install curl -c conda-forge
+        conda install pytorch cpuonly -c pytorch-${{ inputs.pytorch-version }}
+        pip install .[dev,x]
```

Comment:
why is this block somehow replicated with the one below ?

Reply:
conda install for nightly and lts should use `pytorch -c pytorch-<channel here>`, for normal versions is from pytorch channel (`pytorch=<version here> -c pytorch`)